pkg/semtech-loramac/contrib/semtech_loramac.c

```diff
@@ -0,0 +1,624 @@
+                DEBUG("[semtech-loramac] possibly lost interrupt.\n");
+            }
+            break;
+
+        case NETDEV_EVENT_TX_COMPLETE:
+            sx127x_set_sleep((sx127x_t *)dev);
+            semtech_loramac_radio_events.TxDone();
+            DEBUG("[semtech-loramac] Transmission completed\n");
+            break;
+
+        case NETDEV_EVENT_TX_TIMEOUT:
+            msg.type = MSG_TYPE_TX_TIMEOUT;
+
+            if (msg_send(&msg, semtech_loramac_pid) <= 0) {
+                DEBUG("[semtech-loramac] TX timeout, possibly lost interrupt.\n");
+            }
+            break;
+
+        case NETDEV_EVENT_RX_COMPLETE:
+        {
+            size_t len;
+            len = dev->driver->recv(dev, NULL, 0, 0);
+            dev->driver->recv(dev, _semtech_loramac_radio_payload, len, &packet_info);
+            semtech_loramac_radio_events.RxDone(_semtech_loramac_radio_payload,
+                                                 len,
+                                                 packet_info.rssi,
+                                                 packet_info.snr);
+            break;
+        }
+
+        case NETDEV_EVENT_RX_TIMEOUT:
+            msg.type = MSG_TYPE_RX_TIMEOUT;
+
+            if (msg_send(&msg, semtech_loramac_pid) <= 0) {
+                DEBUG("[semtech-loramac] RX timeout, possibly lost interrupt.\n");
+            }
+            break;
+
+        case NETDEV_EVENT_CRC_ERROR:
+            DEBUG("[semtech-loramac] RX CRC error\n");
+            semtech_loramac_radio_events.RxError();
+            break;
+
+        case NETDEV_EVENT_FHSS_CHANGE_CHANNEL:
+            DEBUG("[semtech-loramac] FHSS channel change\n");
+            semtech_loramac_radio_events.FhssChangeChannel(((sx127x_t *)dev)->_internal.last_channel);
+            break;
+
+        case NETDEV_EVENT_CAD_DONE:
+            DEBUG("[semtech-loramac] test: CAD done\n");
+            semtech_loramac_radio_events.CadDone(((sx127x_t *)dev)->_internal.is_last_cad_success);
+            break;
+
+        default:
+            DEBUG("[semtech-loramac] unexpected netdev event received: %d\n",
+                  event);
+    }
+}
+
+void *_semtech_loramac_event_loop(void *arg)
+{
+    (void) arg;
+    static msg_t _msg_q[SEMTECH_LORAMAC_MSG_QUEUE];
+    msg_init_queue(_msg_q, SEMTECH_LORAMAC_MSG_QUEUE);
+    LoRaMacPrimitives_t primitives;
+    LoRaMacCallback_t callbacks;
+
+    _init_loramac(&primitives, &callbacks);
+    semtech_loramac_set_dr(LORAMAC_DEFAULT_DR);
+    semtech_loramac_set_adr(LORAMAC_DEFAULT_ADR);
+    semtech_loramac_set_public_network(LORAMAC_DEFAULT_PUBLIC_NETWORK);
+    semtech_loramac_set_class(LORAMAC_DEFAULT_DEVICE_CLASS);
+
+    while (1) {
+        msg_t msg;
+        msg_receive(&msg);
+        switch (msg.type) {
+            case MSG_TYPE_ISR:
+            {
+                netdev_t *dev = msg.content.ptr;
+                dev->driver->isr(dev);
+                break;
+            }
```

Comment:
here and at other places: I think with a closing brace, the empty line after is not helping readability, so please remove them